packages/twenty-server/src/workspace/workspace-query-runner/factories/arg-setters.factory.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable } from '@nestjs/common';
+
+import { FieldMetadataInterface } from 'src/metadata/field-metadata/interfaces/field-metadata.interface';
+import { WorkspaceQueryRunnerOptions } from 'src/workspace/workspace-query-runner/interfaces/query-runner-option.interface';
+import { ObjectMetadataInterface } from 'src/metadata/field-metadata/interfaces/object-metadata.interface';
+
+import { WorkspaceDataSourceService } from 'src/workspace/workspace-datasource/workspace-datasource.service';
+import { FieldMetadataType } from 'src/metadata/field-metadata/field-metadata.entity';
+
+@Injectable()
+export class ArgsSettersFactory {
+  constructor(
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+  ) {}
+
+  async create(
+    args: Record<string, any>,
+    options: WorkspaceQueryRunnerOptions,
+  ) {
+    const fieldMetadataCollection = options.fieldMetadataCollection;
+
+    const fieldMetadataMap = new Map(
+      fieldMetadataCollection.map((fieldMetadata) => [
+        fieldMetadata.name,
+        fieldMetadata,
+      ]),
+    );
+
+    return this.applySettersToArgsRecursive(args, options, fieldMetadataMap);
+  }
+
+  private async applySettersToArgsRecursive(
```

Comment:
Not sure of the naming too, maybe `computeArgs` ?

Reply:
renamed `createArgsRecursive`